Content.Server/Construction/ConstructionSystem.Graph.cs

```diff
@@ -298,9 +299,20 @@ public bool ChangeNode(EntityUid uid, EntityUid? userUid, string id, bool perfor
                 throw new Exception("Missing construction components");
             }
 
+            // Skip if the new entity's prototype is the same as the original, or the prototype is invalid
             if (newEntity == metaData.EntityPrototype?.ID || !_prototypeManager.HasIndex<EntityPrototype>(newEntity))
                 return null;
 
+            // Optional skip if the new entity's prototype is a parent of the original
+            if (GetCurrentNode(uid, construction)?.ChildrenIgnoreEntity == true &&
+                metaData.EntityPrototype?.ID != null)
+            {
+                var parents = _prototypeManager.EnumerateParents<EntityPrototype>(metaData.EntityPrototype.ID)?.ToList();
+
+                if (parents != null && parents.Any(x => x.ID == newEntity))
+                    return null;
+            }
```

Comment:
this seems extremely weird. What purpose is this even serving? Can you not simply write the construction graph in a way to avoid this?

Reply:
See my comment below